src/tfr_navigation/src/navigator.cpp

```diff
@@ -0,0 +1,135 @@
+#include <navigator.h>
+
+/**
+ *  constructs the sever and binds it to it's execution callback
+ *  displays set parameters and warnings to the user
+ * */
+Navigator::Navigator(ros::NodeHandle &n,std::string name) : node{n},  server{n, name,
+    boost::bind(&Navigator::navigate, this, _1) ,false}, action_name{name},
+    goal_manager()
+{
+    ROS_DEBUG("Navigation server constructed %f", ros::Time::now().toSec());
+    //get parameters
+    ros::param::param<std::string>("~odometry_topic", odometry_topic,
+            "/fused_odom");
+    ros::param::param<float>("~rate", rate, 1);
+    ros::param::param<std::string>("~frame_id", frame_id, "base_footprint");
```

Comment:
Is there a reason these parameters all begin with a `~` character?

Reply:
yes because they are in our private namespace.